File: examples/Fortran/Data_Management_wrong_example.f90
program wrong_usage
    use iso_fortran_env, only : INT32, REAL64
    use openacc
    implicit none
        
    integer(kind=INT32 ), parameter              :: system_size = 10000
    real   (kind=real64), dimension(system_size) :: a, b, c
    integer(kind=INT32 )                         :: i

!  Structured data region
    !$acc data create(a, b) copyout(c)

        !$acc parallel loop present(a(:), b(:))
        do i = 1, system_size
            a(i) = i
            b(i) = i*2
        enddo

        ! We update an element of the array on the CPU
        a(12) = 42

        !$acc parallel loop present(b(:), c(:)) copyin(a(:))
        do i = 1, system_size
            c(i) = a(i) + b(i)
        enddo
    !$acc end data
    write(0,"(a22,f10.5)") "value at position 12: ", c(12)
end program wrong_usage

File: examples/C/Data_Management_wrong_example.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    int size = 10000;
    int a[size], b[size], c[size];

// Structured data region
    #pragma acc data create(a, b) copyout(c)
    {
        #pragma acc parallel loop present(a, b)
        for(int i=0; i<size; ++i){
            a[i] = i;
            b[i] = 2*i;
        }
        // We update an element of the array on the CPU 
        a[14] = 78324;

        #pragma acc parallel loop present(b, c) copyin(a)
        for (int i=0; i<size; ++i){
            c[i] = a[i]+b[i];
        }
    }
    printf("value at position 14: %d\n", c[14]); 
}

